server/src/main/java/org/opensearch/ratelimitting/admissioncontrol/enums/AdmissionControlActionType.java

```diff
@@ -33,13 +34,12 @@ public String getType() {
 
     public static AdmissionControlActionType fromName(String name) {
         name = name.toLowerCase(Locale.ROOT);
-        switch (name) {
-            case "indexing":
-                return INDEXING;
-            case "search":
-                return SEARCH;
-            default:
-                throw new IllegalArgumentException("Not Supported TransportAction Type: " + name);
+
+        for (AdmissionControlActionType type : AdmissionControlActionType.values()) {
```

Comment:
would switch case be more performant here?

Reply:
There are only three values and it shouldn't have difference. But i can revert back as the types are not expected to change.